src/jit/codegenlegacy.cpp

```diff
@@ -19041,7 +19041,10 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                             indCallReg = regSet.rsGrabReg(RBM_ALLINT); // Grab an available register to use for the CALL
                                                                        // indirection
                             instGen_Set_Reg_To_Imm(EA_HANDLE_CNS_RELOC, indCallReg, (ssize_t)addr);
-                            getEmitter()->emitIns_R_R_I(INS_ldr, EA_PTRSIZE, indCallReg, indCallReg, 0);
+#ifdef _TARGET_ARM_
+                            if (!compiler->opts.ShouldUsePInvokeHelpers())
```

Comment:
Sorry, I meant to say:

You should fix this by calling `getAddressOfPInvokeTarget` here and then use `lookup.accessType` to figure out the number of indirections to use - like it is done in RyuJIT backend.

> return &pLookup.addr;

I do not think that this would work as written - it would return address of the `pLookup` local variable.